build/package/Dockerfile.java-toolset

```diff
@@ -0,0 +1,31 @@
+FROM registry.access.redhat.com/ubi8/openjdk-11
+
+SHELL ["/bin/bash", "-o", "pipefail", "-c"]
+
+ENV GRADLE_VERSION=7.2
+
+ARG GRADLE_DOWNLOAD_SHA256=f581709a9c35e9cb92e16f585d2c4bc99b2b1a5f85d2badbd3dc6bff59e1e6dd
+
+ARG GRADLE_WRAPPER_DOWNLOAD_SHA256=f581709a9c35e9cb92e16f585d2c4bc99b2b1a5f85d2badbd3dc6bff59e1e6dd
+
+USER root
+
+RUN cd /opt && \
+    curl -LO https://services.gradle.org/distributions/gradle-${GRADLE_VERSION}-bin.zip && \
+    echo "Checking hash of downloaded gradle distribution" && \
+    echo "${GRADLE_DOWNLOAD_SHA256} gradle-${GRADLE_VERSION}-bin.zip" | sha256sum -c - && \
+    unzip -d /opt/gradle gradle-${GRADLE_VERSION}-bin.zip && \
+    ln -s /opt/gradle/gradle-${GRADLE_VERSION}/bin/gradle /usr/local/bin/gradle && \
+    rm gradle-${GRADLE_VERSION}-bin.zip && \
+    gradle -v && \
+    echo "Loading gradle cache with gradlew  ${GRADLE_VERSION} distribution" && \
+    mkdir -p /tmp/temp-gradle-app && cd /tmp/temp-gradle-app && touch settings.gradle && \
+    gradle wrapper --gradle-distribution-sha256-sum ${GRADLE_WRAPPER_DOWNLOAD_SHA256} && ./gradlew -version && \
+    mkdir -p /home/gradle && \
+    ln -s /root/.gradle /home/gradle/.gradle && \
+    chmod -R 777 /home/gradle/.gradle && \
```

Comment:
Would be nice to figure out why `777` is used here. I suspect we do not need this permissions.

Reply:
I'll verify if 777 is needed.

Reply:
This is not needed at the moment. The user running the task and therefore the one used when `build-java.sh` is called is the user `root`. It has access to the folder.

If in openshift the user running the task is a different one then `gradlew` will likely not have access to `/home/gradle/.gradle`.

The benefit of pre-loading the gradle cache folder is that calling `gradlew` will not require to download the gradle distribution because it is already in the cache.

BTW: to improve performance of java/gradle builds we should explore to set the gradle cache to point to a persistent storage. Otherwise each build will need to download everytime all dependencies.

Reply:
> If in openshift the user running the task is a different one then gradlew will likely not have access to /home/gradle/.gradle.

We need to ensure that the root group (gid=0) has enough permissions. The random user in OpenShift is in the root group.

> to improve performance of java/gradle builds we should explore to set the gradle cache to point to a persistent storage. Otherwise each build will need to download everytime all dependencies.

See #147. We already have a PVC which backs the pipeline, so in theory we can use this to preserve the cache. However, at the moment `ods-start` wipes the working directory. We could change this as we control that task. FYI @oalyman I know you worked on this in Jenkins - would be great to add this to Tekton. Maybe we can have a quick discussion what the best approach is?

Reply:
to not forget this important one I opened #212 to add add a PVC as build cache